torch/csrc/jit/python/pybind_utils.h

```diff
@@ -948,10 +950,71 @@ inline py::object runAndInsertCall(
   return toPyObject(std::move(stack.back()));
 }
 
+inline c10::optional<py::object> maybeTorchFunctionDispatch(
+    const py::object& callee,
+    const tuple_slice& args_no_self,
+    const py::kwargs& kwargs,
+    const c10::QualifiedName qualname) {
+  std::vector<py::handle> args_vec = {callee};
+  for (const auto& arg : args_no_self) {
+    args_vec.push_back(arg);
+  }
+  py::tuple args = py::cast(args_vec);
+
+  // Handle __torch_function__ dispatch
+  std::vector<py::handle> overloaded_args;
+  size_t total_arg_num = args.size() + kwargs.size();
+  for (size_t i = 0; i < args.size(); ++i) {
+    is_tensor_and_append_overloaded(args[i].ptr(), &overloaded_args);
```

Comment:
This will fail for `TensorList` operands.

Reply:
Can this take `TensorList`s? I believe the reason to skip in the other place was simply that they didn’t need it.

Reply:
I see from [this comment](https://github.com/pytorch/pytorch/pull/52884#issuecomment-788128434) that no `TensorList`s are needed, if so, you can resolve this, @jamesr66a.

Reply:
@hameerabbasi  I don't really understand this comment thread.

1) Can you explain exactly what would fail about `TensorList`? I didn't write this code, but from what I can tell, `is_tensor_and_append_overloaded` and `is_tensor_list_and_append_overloaded` are both best-effort and both if `args` is a `Tensor` or if `arg` is a `TensorList` should be covered by the two cases here. Can you explain a bit more about what you're referring to?
2) I don't understand the implication of Ed's comment that suggests that `TensorList` operands are not possible.

Reply:
`is_tensor_and_append_overloaded` never checks for `TensorList`, only for something that "looks like a `Tensor`", i.e. has `__torch_function__` and isn't `nn.Parameter` or `Tensor`. See this comment: https://github.com/pytorch/pytorch/blob/a0a1bb074b954338d7f19dac9a9c8136376bce7c/torch/csrc/utils/python_arg_parser.h#L748-L757

If `arg` turns out to be a `TensorList`, then it won't recurse inside the `TensorList` and therefore miss types that have `__torch_function__`.

What I'm looking for here is confirmation that `arg` will never be a `TensorList`, i.e. that the callable that you're overriding here will never take a `TensorList` as a potential argument. Ed's comment has an example of what the callable you're overriding might look like, and if it only takes a `Tensor` in, then we're okay.

Reply:
@hameerabbasi still don't understand your comment. In this code, there's a call to `is_tensor_list_and_append_overloaded` directly after this comment thread, which AFAICT does do this recursion.

Reply:
Ah, missed that one. I apologize for the confusion; consider this resolved.